Dockerfile.lambda

```diff
@@ -0,0 +1,19 @@
+FROM golang:1.17 AS build
+
+# See https://github.com/buildkite/buildkite-agent-metrics/releases for latest version
+ENV LAMBDA_VERSION="5.3.0" 
+
+WORKDIR /app
+
+RUN apt-get update && apt-get install -y wget tar && \
+    wget "https://github.com/buildkite/buildkite-agent-metrics/archive/refs/tags/v${LAMBDA_VERSION}.tar.gz" && \
+    tar xvzf "v${LAMBDA_VERSION}.tar.gz" && \
+    cd "buildkite-agent-metrics-${LAMBDA_VERSION}" && \
+    go get ./lambda && \
+    CGO_ENABLED=0 go build -o ../handler ./lambda
```

Comment:
we might be able to change this to use 
```
wget https://github.com/buildkite/buildkite-agent-metrics/releases/latest/download/handler.zip
```

and avoid actually having to build the binary as part of the docker build process, as well as removing the need for a static `LAMBDA_VERSION` envar

Reply:
I can fix it up to do this.

Reply:
sounds great :)